verticapy/learn/delphi.py

```diff
@@ -0,0 +1,835 @@
+            modeltype = None
+            if not(isinstance(input_relation, vDataFrame)):
+                vdf = vdf_from_relation(input_relation, cursor=cursor)
+            else:
+                vdf = input_relation
+            if self.parameters["estimator_type"].lower() == "binary" or (self.parameters["estimator_type"].lower() == "auto" and sorted(vdf[y].distinct()) == [0, 1]):
+                self.parameters["estimator_type"] = "binary"
+                if self.parameters["estimator"] == "native":
+                    self.parameters["estimator"] = [LinearSVC(self.name, cursor=self.cursor), LogisticRegression(self.name, cursor=self.cursor), RandomForestClassifier(self.name, cursor=self.cursor), XGBoostClassifier(self.name, cursor=self.cursor), NaiveBayes(self.name, cursor=self.cursor)]
+                elif self.parameters["estimator"] == "fast":
+                    self.parameters["estimator"] = [LogisticRegression(self.name, cursor=self.cursor), RandomForestClassifier(self.name, cursor=self.cursor), NaiveBayes(self.name, cursor=self.cursor)]
+                else:
+                    self.parameters["estimator"] = [LinearSVC(self.name, cursor=self.cursor), LogisticRegression(self.name, cursor=self.cursor), RandomForestClassifier(self.name, cursor=self.cursor), XGBoostClassifier(self.name, cursor=self.cursor), NaiveBayes(self.name, cursor=self.cursor), KNeighborsClassifier(self.name, cursor=self.cursor), NearestCentroid(self.name, cursor=self.cursor)]
+            elif self.parameters["estimator_type"].lower() == "regressor" or (self.parameters["estimator_type"].lower() == "auto" and vdf[y].isnum()):
+                self.parameters["estimator_type"] = "regressor"
+                if self.parameters["estimator"] == "native":
+                    self.parameters["estimator"] = [LinearSVR(self.name, cursor=self.cursor), ElasticNet(self.name, cursor=self.cursor), LinearRegression(self.name, cursor=self.cursor), Lasso(self.name, cursor=self.cursor), Ridge(self.name, cursor=self.cursor), RandomForestRegressor(self.name, cursor=self.cursor), XGBoostRegressor(self.name, cursor=self.cursor),]
+                elif self.parameters["estimator"] == "fast":
+                    self.parameters["estimator"] = [LinearRegression(self.name, cursor=self.cursor), ElasticNet(self.name, cursor=self.cursor), RandomForestRegressor(self.name, cursor=self.cursor),]
+                else:
+                    self.parameters["estimator"] = [LinearSVR(self.name, cursor=self.cursor), ElasticNet(self.name, cursor=self.cursor), LinearRegression(self.name, cursor=self.cursor), Lasso(self.name, cursor=self.cursor), Ridge(self.name, cursor=self.cursor), RandomForestRegressor(self.name, cursor=self.cursor), XGBoostRegressor(self.name, cursor=self.cursor), KNeighborsRegressor(self.name, cursor=self.cursor),]
+            elif self.parameters["estimator_type"].lower() in ("multi", "auto",):
```

Comment:
It seems that "auto" is always treated the same as "multi"! Then, how have you achieved auto detection?

Reply:
It's an 'elif', you should look at the previous one.